packages/rum-core/src/domain/vital/vitalCollection.ts

```diff
@@ -9,61 +9,85 @@ import { PageState } from '../contexts/pageStateHistory'
 
 export interface DurationVitalStart {
   name: string
-  startClocks: ClocksState
   context?: Context
+  details?: string
 }
 
 export interface DurationVitalStop {
+  context?: Context
+  details?: string
+}
+
+export interface DurationVitalAdd {
   name: string
-  stopClocks: ClocksState
+  startClocks: ClocksState
+  duration: Duration
   context?: Context
+  details?: string
+}
+
+export interface DurationVitalInstance {
+  stop: (options?: DurationVitalStop) => void
 }
 
 interface DurationVital {
   name: string
   type: VitalType.DURATION
   startClocks: ClocksState
-  value: Duration
+  duration: Duration
+  details?: string
   context?: Context
 }
 
 export function startVitalCollection(lifeCycle: LifeCycle, pageStateHistory: PageStateHistory) {
-  const vitalStartsByName = new Map<string, DurationVitalStart>()
-
-  lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
-    // Discard all the vitals that have not been stopped to avoid memory leaks
-    vitalStartsByName.clear()
-  })
-
   function isValid(vital: DurationVital) {
-    return !pageStateHistory.wasInPageStateDuringPeriod(PageState.FROZEN, vital.startClocks.relative, vital.value)
+    return !pageStateHistory.wasInPageStateDuringPeriod(PageState.FROZEN, vital.startClocks.relative, vital.duration)
   }
 
   return {
-    startDurationVital: (vitalStart: DurationVitalStart) => {
-      vitalStartsByName.set(vitalStart.name, vitalStart)
-    },
-    stopDurationVital: (vitalStop: DurationVitalStop) => {
-      const vitalStart = vitalStartsByName.get(vitalStop.name)
-      if (!vitalStart) {
-        return
-      }
-      const vital = buildDurationVital(vitalStart, vitalStop)
-      vitalStartsByName.delete(vital.name)
+    addDurationVital: (vitalAdd: DurationVitalAdd) => {
+      const vital = assign({ type: VitalType.DURATION }, vitalAdd)
+
       if (isValid(vital)) {
         lifeCycle.notify(LifeCycleEventType.RAW_RUM_EVENT_COLLECTED, processVital(vital, true))
       }
     },
   }
 }
 
-function buildDurationVital(vitalStart: DurationVitalStart, vitalStop: DurationVitalStop) {
+export function createVitalInstance(
+  cb: (vital: DurationVital) => void,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Could be nice to have an more explicite name 
```suggestion
  stopCallback: (vital: DurationVital) => void,
```